src/constant_value.rs

```diff
@@ -56,21 +60,25 @@ impl ConstantValue {
 
 /// Keeps track of MIR constants that have already been mapped onto ConstantValue instances.
 pub struct ConstantValueCache {
+    char_cache: HashMap<char, ConstantValue>,
```

Comment:
type CacheMap<T> = HashMap<T, ConstantValue>;

above the struct would reduce a little bit of this repetition. But I know you prefer things to be direct and clear.

Reply:
I'm using separate tables to avoid possible hash collision between different types of keys. Given that, HashMap<char, ConstantValue> is just a wee bit more obvious than CacheMap<char> not least because the latter name is at once non standard and quite vague/general.